### PR TITLE
File#path raises an IOError for files opened with File::Constants::TMPFILE option

### PR DESCRIPTION
**Pull request by andrykonchin**
https://bugs.ruby-lang.org/issues/13568

Related tests
https://github.com/ruby/ruby/commit/75cda5e22f6bc7ba6c91b6501d4233e782178a3a
https://github.com/ruby/ruby/blob/ccbd6ee55ab1ca169fbc69f2283c812343e8387f/test/ruby/test_file.rb#L493

### CODE REVIEW

**Comment by andrykonchin** [comment]
Looks like CI is broken

```
Worker information
...
oci runtime error: exec failed: container_linux.go:265: starting container process caused "could not create session key: disk quota exceeded"
```

File: core/file/shared/path.rb [hunk 1]
```diff
@@ -15,6 +15,13 @@
     @file.send(@method).should be_an_instance_of(String)
   end
 
+  it "calls to_str on argument and returns exect value" do
```
Comment (eregon) [inline core/file/shared/path.rb]: `exact`

File: core/file/path_spec.rb [hunk 2]
```diff
@@ -3,6 +3,32 @@
 
 describe "File#path" do
   it_behaves_like :file_path, :path
+
+  ruby_version_is "2.5" do
+    platform_is :linux do
+      if defined?(File::TMPFILE)
```
Comment (eregon) [inline core/file/path_spec.rb]: Can you use `guard -> { defined?(File::TMPFILE) } do` instead?
It allows to easily disable all guards which is useful in some cases (see https://github.com/ruby/mspec/issues/30#issuecomment-312487779 for details).

File: core/file/path_spec.rb [hunk 3]
```diff
@@ -3,6 +3,32 @@
...
+        end
+
+        after :each do
+          rm_r @dir
+        end
+
+        it "raises IOError if file was opened with File::TMPFILE" do
+          begin
+            File.open(@dir, File::RDWR | File::TMPFILE) do |f|
+              -> { f.path }.should raise_error(IOError)
```
Comment (eregon) [inline core/file/path_spec.rb]: What happen if this was `f.to_path`?

**Comment by eregon** [comment]
@andrykonchin I added a few comments. For some reasons they are shown as outdated but I think they still apply.

**Comment by andrykonchin** [comment]
Thanks. Done

**Pull request merged**